inst/www/htmlwidgets.js

```diff
@@ -660,7 +660,16 @@
   }
 
   // Wait until after the document has loaded to render the widgets.
-  if (document.addEventListener) {
+  if (shinyMode && window.jQuery) {
+    // Note: this case is here to ensure that initShiny executes
+    // before staticRender executes. Since initShiny currently
+    // registers important methods like Shiny.onInputChange, if
+    // it happens to execute after staticRender, the widget
+    // won't have access to all of Shiny's methods.
+    window.jQuery(function() {
+      setTimeout(window.HTMLWidgets.staticRender, 1);
```

Comment:
Are you sure this needs a `setTimeout`? I actually don't think it does. I know that leaflet `updateBounds` function needs `init_shiny` to have completed, but for that reason, leaflet itself calls `setTimeout(updateBounds)`

Reply:
Yes, plotly needs it; without it, `Shiny.setInputValue()` won't be available when `HTMLWidgets.renderValue()` is called.

Reply:
@cpsievert And that issue for plotly doesn't repro with the CRAN version of Shiny?

Reply:
Sorry, I'm not trying to be difficult, just trying to make the most conservative change we can for now.

Reply:
Oh, it does repro with CRAN versions...I wouldn't be surprised if there are lots of htmlwidgets out there are unknowingly suffering from the same issue.

What's the potential danger in deferring `staticRender()`? Is it that shiny might not be able to bind to the widget?